lib/internal/util/inspect.js

```diff
@@ -439,6 +439,82 @@ defineColorAlias('inverse', 'swapColors');
 defineColorAlias('inverse', 'swapcolors');
 defineColorAlias('doubleunderline', 'doubleUnderline');
 
+// TODO(BridgeAR): Consider using a class instead and require the user to instantiate it
+// before using to declare their color support (or use a default, if none
+// provided).
```

Comment:
Why wouldn’t or couldn’t these be simple static functions that require no initialization? I mean, aren’t they essentially just prepending and appending `\u001b[90m` and `\u001b[39m` and the like?

As a user, I’d prefer they be as terse as possible. I’d be more likely to use them the simpler they are.

Reply:
Yes and no: if we add more colors such as hex colors later on, we'll need to check that the terminal does indeed support the colors. Currently there's no way to do this besides a best effort check and we sometimes detect colors when not being supported and the other way around. Therefore I would like to provide a way to define the support by the user:

```js
const { Colorize } = require('formatter')

const colorize256 = new Colorize({ colors: 256 })

console.log(
  colorize256.bold.underline.hex('#123456')('bold underline fancy color')
)
```

Using `Colorize.bold.underline.hex('#123456')('bold underline maybe fancy color')` could try to auto detect the supported colors while using the instance would be fixed.

Reply:
These feel like almost two use cases: a simple CLI tool like a test runner that needs only a handful of colors (or only green and red); and a more complicated CLI tool that wants lots of colors. The latter can call some kind of initialization function, since it’s somewhat expected that more work would be required to do something fancy; but ideally the simple case wouldn’t need much ceremony.